File: 프로그래머스/lv1_문자열_내_마음대로_정렬하기.py
def solution(strings, n):
    answer = []

    answer = sorted(strings, key = lambda x : [x[n], x])
    # strings.sort(key = lambda x: x[n]+x)
    '''
        key를 기준으로 sort 하도록 함
    '''

    return answer

if __name__ == "__main__":
    strings = ["abce", "abcd", "cdx"]
    n = 2
    ret = solution(strings, n)
    print(ret)

File: 프로그래머스/lv1_문자열_내_마음대로_정렬하기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int global_n;

bool cmp(string a, string b){

    if (a[global_n] < b[global_n])
        return true;
    else if ( a[global_n] == b[global_n] ){
        if( a <= b ){
            return true;
        }
        else {
            return false;
        }
    }
    else{
        return false;
    }
}

vector<string> solution(vector<string> strings, int n) {
    vector<string> answer;
    global_n = n;

    sort(strings.begin(), strings.end(), cmp);

    answer = strings;

    return answer;
}

int main(void){

    vector<string> strings = {"abce", "abcd", "cdx"};
    int n = 1;

    vector<string> ret;
    ret = solution(strings, n);

    for( string elem : ret ){
        cout << elem << " ";
    }

    return 0;
}